modules/foreman_search_facts.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2018, Sean O'Keeffe <seanokeeffe797@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: foreman_search_facts
```

Comment:
I don't like the word `facts` here, because it makes me think of puppet. What about just `foreman_search`?

Reply:
I guess, i was the one to introduce the name *_facts. I was inspired by the vmware_*_facts modules. I do not know, whether it is the best way. I even thought about not making a separate module at all, just returning the entity if there was nothing specified to change it.

Reply:
We already have `foreman_setting_facts`. I guess one day we might have `foreman_puppet_facts`. All that makes sense to me, so I withdraw my previous comment. :fax: